### ISSUE TITLE
Follow HLint suggestions?

### ISSUE THREAD

**Issue by philderbeast** [gelisam/klister#233]
Running `hlint-3.8`, I'm finding 148 hints. Are you open to following these suggestions? If so then I'd be happy to contribute those changes.

- [ ] Avoid lambda, 1 hint
- [ ] Eta reduce, 12 hints
- [ ] Fuse foldr/map, 1 hint
- [ ] Move brackets to avoid $, 6 hints
- [ ] Redundant $, 32 hints
- [ ] Redundant <$>, 17 hints
- [ ] Redundant bracket, 12 hints
- [ ] Replace case with fromMaybe, 3 hints
- [ ] Use $>, 2 hints
- [ ] Use <&>, 1 hint
- [ ] Use >=>, 1 hint
- [ ] Use asks, 5 hints
- [ ] Use camelCase, 2 hints
- [ ] Use const, 4 hints
- [ ] Use evalState, 2 hints
- [ ] Use fold, 1 hint
- [ ] Use for, 1 hint
- [ ] Use fromMaybe, 1 hint
- [ ] Use id, 6 hints
- [ ] Use newtype instead of data, 3 hints
- [ ] Use record patterns, 4 hints
- [ ] Use tuple-section, 2 hints
- [ ] Use uncurry, 2 hints
- [ ] Use unless, 3 hints
- [ ] Use unwords, 1 hint
- [ ] Use void, 5 hints
- [ ] Use when, 2 hints

**Comment by doyougnu** [comment]
Yes please! We are thankful for the help. Most of these should be extremely minor changes. The only one that will be difficult and that I think you should avoid are the incomplete pattern matches in the Evaluator. Fixing these will mean re-architecting the Evaluator and is just not worth it at this time.

**Comment by philderbeast** [comment]
> I seem to have a different version of hlint than you, because my hlint report only finds one hint

Yes that is likely. If we go for an action, the hlint version can be set there.

**Comment by gelisam** [comment]
Ah, so if we merge #234, hlint will not pester us about any of the suggestions listed there! I had misunderstood the purpose of that list, I thought it was the opposite. Sure, let's merge it!
